Ginger/Ginger/UserControlsLib/UCDataMapping.xaml.cs

```diff
@@ -0,0 +1,553 @@
+
+            MarkMappedValueValidation();
+        }
+        
+        private void DisableAllTypeOptions()
+        {
+            GingerCore.General.DisableComboItem(xMappedTypeComboBox, eDataType.Variable);
+            GingerCore.General.DisableComboItem(xMappedTypeComboBox, eDataType.OutputVariable);
+            GingerCore.General.DisableComboItem(xMappedTypeComboBox, eDataType.GlobalVariable);
+            GingerCore.General.DisableComboItem(xMappedTypeComboBox, eDataType.ApplicationModelParameter);
+            GingerCore.General.DisableComboItem(xMappedTypeComboBox, eDataType.DataSource);
+            xDSConfigBtn.IsEnabled = false;
+        }
+        
+        public static DataTemplate GetTemplate(string dataTypeProperty, string dataValueProperty, string enableDataMappingProperty = "", string variabelsSourceProperty = "", ObservableList<string> variabelsSourceList=null, string outputVariabelsSourceProperty = "")
+        {
+            DataTemplate template = new DataTemplate();
+            FrameworkElementFactory ucDataMapping = new FrameworkElementFactory(typeof(UCDataMapping));
+
+            if (string.IsNullOrEmpty(variabelsSourceProperty) == false)
+            {
+                Binding variablesSourceBinding = new Binding(variabelsSourceProperty);
+                variablesSourceBinding.Mode = BindingMode.OneWay;
+                variablesSourceBinding.UpdateSourceTrigger = UpdateSourceTrigger.PropertyChanged;
+                ucDataMapping.SetBinding(UCDataMapping.VariabelsSourceProperty, variablesSourceBinding);
+            }
+            else if(variabelsSourceList != null)
+            {
+                ucDataMapping.SetValue(UCDataMapping.VariabelsSourceProperty, variabelsSourceList);
+            }
+
+            if (string.IsNullOrEmpty(outputVariabelsSourceProperty) == false)
+            {
+                Binding outputVariabelsSourceBinding = new Binding(outputVariabelsSourceProperty);
+                outputVariabelsSourceBinding.Mode = BindingMode.OneWay;
+                outputVariabelsSourceBinding.UpdateSourceTrigger = UpdateSourceTrigger.PropertyChanged;
+                ucDataMapping.SetBinding(UCDataMapping.OutputVariabelsSourceProperty, outputVariabelsSourceBinding);
+            }
+
+            Binding mappedItemTypeBinding = new Binding(dataTypeProperty);
+            mappedItemTypeBinding.Mode = BindingMode.TwoWay;
+            mappedItemTypeBinding.UpdateSourceTrigger = UpdateSourceTrigger.PropertyChanged;
+            ucDataMapping.SetBinding(UCDataMapping.MappedTypeProperty, mappedItemTypeBinding);
+
+            Binding mappedValueBinding = new Binding(dataValueProperty);
+            mappedValueBinding.Mode = BindingMode.TwoWay;
+            mappedValueBinding.UpdateSourceTrigger = UpdateSourceTrigger.PropertyChanged;
+            ucDataMapping.SetBinding(UCDataMapping.MappedValueProperty, mappedValueBinding);
+
+            if (string.IsNullOrEmpty(enableDataMappingProperty) == false)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5538205)